clientdb/order.go

```diff
@@ -621,11 +633,73 @@ func DeserializeOrder(nonce order.Nonce, r io.Reader) (
 // supplied reader by interpreting it as a tlv stream. If successful any
 // non-default values of the additional data will be set on the given order.
 func deserializeOrderTlvData(r io.Reader, o order.Order) error {
+	var (
+		sidecarTicket []byte
+	)
+
+	// We'll add records for all possible additional order data fields here
+	// but will check below which of them were actually set, depending on
+	// the order type as well.
+	tlvStream, err := tlv.NewStream(
+		tlv.MakePrimitiveRecord(sidecarTicketType, &sidecarTicket),
+	)
+	if err != nil {
+		return err
+	}
+
+	parsedTypes, err := tlvStream.DecodeWithParsedTypes(r)
+	if err != nil {
+		return err
+	}
+
+	// Now check what records were actually parsed from the stream and
+	// assign any parsed fields to our order.
+	switch castOrder := o.(type) {
+	case *order.Ask:
+
+	case *order.Bid:
+		if t, ok := parsedTypes[sidecarTicketType]; ok && t == nil {
```

Comment:
Maybe I misread the documentation here, but do we need to check `t == nil`? Seems like `t != nil && ok` would be an error.

Reply:
So if the type is found and it's known, it'll be in the map but with a nil value (`ok` check is important here). If it's found and the type is _unknown_ then it'll be in the map, but the value will be the raw bytes of the type.